codegenerator/cli/src/config_parsing/system_config.rs

```diff
@@ -447,6 +449,46 @@ impl SystemConfig {
         })
     }
 
+    fn load_env_vars(project_root: &PathBuf) -> Result<EnvMap> {
+        let env_map = match EnvLoader::with_path(project_root.join(".env"))
+            .sequence(EnvSequence::InputThenEnv)
+            .load()
+        {
+            Ok(env_map) => env_map,
+            // It'll fail if .env is missing, so we return a map with std::env only
+            Err(dotenvy::Error::Io(_, _)) => EnvLoader::new()
+                .sequence(EnvSequence::EnvOnly)
+                .load()
+                .context("Failed loading environment variables")?,
+            err => err.context("Failed loading .env file")?,
+        };
+        Ok(env_map)
+    }
+
+    fn interpolation_config_variables(config_string: String, env_map: EnvMap) -> Result<String> {
+        let mut missing_vars = Vec::new();
+
+        let re = Regex::new(r"\$\{([a-zA-Z_][0-9a-zA-Z_]*)\}").unwrap();
```

Comment:
TBH, I don't love regex. Because I never fully understand it 😓 much prefer parser combinator approach but I do understand how useful regex is.

If regex is used I'd prefer it to be moved into it's own function with unit tests specifically for that function.

Questions I have that tests would solve:
1. What are the boundaries of the match? Looks like for instance that inside the braces it won't accept symbols other than `_`? That I'm even not sure of... so would be good to show acceptable/unacceptable values in a test.
2. What if it doesn't match the inner braces but the user has added ${}? Looks like we don't have a good error to help the user with that.
3. How does it handle escaping? Not sure where but maybe I want to escape `\${` in my config. etc.

Reply:
1. `[]` is something like or in the RegExp world, so it'll either a-z, A-Z or _
2. Yes, indeed
3. This is currently not supported. We can add support for it later when somebody requests

Reply:
I personally used a parser combinator just a several times and find Regex much simpler. But this is probably because my first ever Js app heavely used regex to parse complex HTML 😅